analysis_options.yaml

```diff
@@ -1,13 +1,37 @@
 
-include: package:lints/recommended.yaml
+# Lints set up with some guidance from here:
+# https://rydmike.com/blog_flutter_linting.html
+
+include: package:all_lint_rules_community/all.yaml
 
-linter:
-  rules:
-    - always_declare_return_types
-    - prefer_single_quotes
-    - unawaited_futures
-    - avoid_void_async
 
 analyzer:
-#   exclude:
-#     - path/to/excluded/files/**
+  language:
+    strict-casts: true
+    strict-inference: true
+    strict-raw-types: true
+  
+  errors:
+    # Without ignore here, we cause import of all_lint_rules to warn, because some rules conflict.
+    # We explicitly enabled even conflicting rules and are fixing the conflicts in this file.
+    # Put it to warning temporarily, if you need troubleshoot lint rule settings.
+    included_file_warning: ignore
+
+    # Treat missing required parameters as an error, not as a hint or a warning.
+    missing_required_param: error
+
+    # Treat missing returns as an error, not as a hint or a warning.
+    missing_return: error
```

Comment:
In general, this can be removed as it is intended for old code. Instead, it now uses `body_might_complete_normally`, which throws an error by default.

Reply:
thanks, good point!  i realized missing_required_param was also for old code (looks like pre-null safety) so I removed that one too.  now there's no more need for the error section at all!